### PR TITLE
close #47 navbar

### LINKED ISSUE #47: navbar buttons
shows the right buttons on the navbar
- Register page: Login
- Login page: Register
- Spaces page: New space, Requests, Logout
- Requests page: New space, Spaces, Logout
- New space page: Spaces, Requests, Logout
- New request page: Spaces, Requests, Logout

### PR DESCRIPTION
**Pull request by robota-x**
- Buttons are shown only in the pages where they make sense, f.ex. no 'register' link on the register page, and so on
- Please comment if you think different buttons should be displayed on the various pages

### CODE REVIEW

File: app/server.rb [hunk 1]
```diff
@@ -13,6 +13,10 @@ def current_user
       @current_user ||= User.get(session[:user_id])
     end
 
+    def current_path
+      request.path_info
+    end
+
```
Comment (kylebuttner) [inline app/server.rb:19]: This is good to have!

File: app/views/_navbar.erb [hunk 2]
```diff
@@ -5,28 +5,48 @@
     </section>
 
     <section>
+
+      <% unless current_path == '/spaces' %>
+      <form action='/spaces' method="get">
+          <button type="submit">Available spaces</button>
```
Comment (kylebuttner) [inline app/views/_navbar.erb]: Let's call this button **Spaces** as per the spec.

File: app/views/_navbar.erb [hunk 3]
```diff
@@ -5,28 +5,48 @@
     </section>
 
     <section>
+
+      <% unless current_path == '/spaces' %>
+      <form action='/spaces' method="get">
```
Comment (kylebuttner) [inline app/views/_navbar.erb]: Let's indent this form.

File: app/views/_navbar.erb [hunk 4]
```diff
@@ -5,28 +5,48 @@
...
+          <button type="submit">Available spaces</button>
+      </form>
+      <%end%>
+
+      <% if current_path == '/spaces' || current_path == '/requests' %>
```
Comment (kylebuttner) [inline app/views/_navbar.erb]: Let's remove this button from the navbar, move it to `/spaces`, and rename it **List a space** as the spec asks.

File: app/views/_navbar.erb [hunk 5]
```diff
@@ -5,28 +5,48 @@
...
       <form action='/spaces/new' method="get">
           <button type="submit">Create space</button>
       </form>
+      <%end%>
+
+      <% unless current_path == '/requests' %>
       <form action='/requests'>
         <button type="submit">Check requests</button>
```
Comment (kylebuttner) [inline app/views/_navbar.erb]: Let's rename this **Requests** as the spec asks.

File: app/views/_navbar.erb [hunk 6]
```diff
@@ -5,28 +5,48 @@
...
-     <form action='/user/new'>
-       <button type="submit">Register</button>
-     </form>
+      <% unless current_path == '/session/new' %>
+        <form action='/session/new'>
+        <button type="submit">Log in</button>
+        </form>
+      <%end%>
+
+      <% unless current_path == '/user/new' %>
```
Comment (kylebuttner) [inline app/views/_navbar.erb]: I think we should remove the **Register** button in favor of a sign up page on the homepage as the spec requests.

File: spec/features/navbar_spec.rb [hunk 7]
```diff
@@ -1,37 +1,120 @@
...
-    expect(page).not_to have_content('Welcome')
-  end
 
-  scenario 'Login' do
-    visit '/'
-    click_button 'Log in'
-    expect(current_path).to eq '/session/new'
-    expect(page).to have_content("Log in")
-  end
+  feature 'Buttons functionality' do
```
Comment (kylebuttner) [inline spec/features/navbar_spec.rb:3]: If we change the button names above, we will need to change the tests below.

**Comment by kylebuttner** [comment]
Looks good @lorenzoturrino! I just let some comments regarding naming conventions. I think we should rename/remove some of the buttons to be in line with the [spec](https://github.com/makersacademy/course/blob/master/lab_week/makers_bnb_images/MakersBnB_mockups.pdf).

File: app/views/_navbar.erb (hunk 6 above)
Comment (robota-x) [inline app/views/_navbar.erb]: leaving it there for now after consultation to have a 'escape' route from the login page other than a successful login (a new user might click on the login button by mistake)

**Pull request merged**